### PR TITLE
catch exception in option UI

### PR DESCRIPTION
**Pull request by zhili1208**
## Bug
Fixes: VSTS 600427
Regression: No  
If Regression then when did it last work:   
If Regression then how are we preventing it in future:   

## Fix
Details:
we should not throw any exception in UI which crash VS.
catch any exception and log it to activity log.

## Testing/Validation
Tests Added: No  
Reason for not adding tests:  this can be covered by Apex or vendor test, since we are doing apex design review now, considering do vendor test for now, add Apex test in future.
Validation done:  manual validation

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs [hunk 1]
```diff
@@ -62,11 +62,16 @@ internal void OnActivated()
                 catch (InvalidOperationException)
                 {
                     MessageHelper.ShowErrorMessage(Resources.ShowError_ConfigInvalidOperation, Resources.ErrorDialogBoxTitle);
-               } 
+                } 
                 catch (UnauthorizedAccessException)
                 {
                     MessageHelper.ShowErrorMessage(Resources.ShowError_ConfigUnauthorizedAccess, Resources.ErrorDialogBoxTitle);
                 }
+                catch (Exception ex)
```
Comment (rohit21agrawal) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs:70]: i do not agree that we need to catch ALL the exceptions. this is just going to hide any possible exceptions that get thrown from the above code path.

have you been able to determine what exception is being thrown and why?

Comment (zhili1208) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs:70]: any exception from option UI crash VS, we did this three times, every time we caught a specified exception, we got a watson crash later. we should make sure nuget do not crash VS first and it not hides the exception, VS pops up a error window and log the exception to activity log

Comment (rohit21agrawal) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs:70]: are we able to look at the exceptions in activity log the same way we can look at Watson results?

Comment (zhili1208) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs:70]: Yes, we will have full stack in activity log, we already did this for applying setting, just missed the init part before.

Comment (rohit21agrawal) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Options/GeneralOptionControl.cs:70]: no what I mean is  that is the activity log available to us without people filing a feedback ticket?

**Comment by zhili1208** [comment]
Per discussion in standup, we decide to catch the exceptions in UI and tracking those exceptions in our data service 【https://github.com/NuGet/Home/issues/6814】

**Review by rohit21agrawal — APPROVED** [review]

**Comment by rrelyea** [comment]
@zhili1208  - before we merge this fix, can you make it so that we give more actionable errors. Perhaps mention nuget.config -- and mention the full path???  otherwise, how do they know what to go fix??

**Pull request merged**